File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s): 
    answer = ""
    text = ""
    
    num_dic = {
        "zero": 0,
        "one": 1,
        "two": 2,
        "three": 3,
        "four": 4,
        "five": 5,
        "six": 6,
        "seven": 7,
        "eight": 8,
        "nine": 9,
    }
    
    for x in s:
        print(x)
        if x.isdigit():
            answer += x
        else:
            text += x
            if text in num_dic:
                answer += str(num_dic[text])
                text = ""
                
            
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    let answer = "";
    let numStr = "";
    
    const numDic = {
        "zero": 0,
        "one": 1,
        "two": 2,
        "three": 3,
        "four": 4,
        "five": 5,
        "six": 6,
        "seven": 7,
        "eight": 8,
        "nine": 9,
    }
    
    s.split("").forEach((w) => {
        if(!isNaN(w)){
            if(numStr !== "") answer += numDic[numStr];
            answer += w;
            numStr = "";
        }else{
            numStr += w;
            if(numStr in numDic) {
                answer += numDic[numStr];
                numStr = "";
            }
            
        }
    });
    
    if(numStr !== "") answer += numDic[numStr];
    
    return parseInt(answer);
}